vdev/src/util.rs

```diff
@@ -37,15 +37,28 @@ pub fn git_head() -> Result<Output> {
         .context("Could not execute `git`")
 }
 
-/// Calculate the release channel from `git describe`
-pub fn release_channel() -> Result<&'static str> {
-    git_head().map(|output| {
-        if output.status.success() {
-            "latest"
-        } else {
-            "nightly"
+// If in CI, it is necessary to adjust the permissions on the git repository
+// as a precursor to running some git commands.
+pub fn mark_safe_git_repo() {
+    if let Ok(is_ci) = std::env::var("CI") {
+        if is_ci == "true" {
+            std::process::Command::new("git")
+                .args([
+                    "config",
+                    "--global",
+                    "--add",
+                    "safe.directory",
+                    "/git/vectordotdev/vector",
+                ])
+                .output()
+                .context("Could not execute `git config --add safe.directory`")
+                .unwrap();
         }
-    })
+    }
+}
+
+pub fn get_mode() -> String {
+    std::env::var("MODE").unwrap_or_else(|_| "custom".to_string())
```

Comment:
I'd prefer this to be named something other than just "mode" and `$MODE`. What kind of mode? We refer to it as "release channel" elsewhere, maybe use that?